src/main/java/edu/umich/tl/analytics/CanvasCourseReportServlet.java

```diff
@@ -46,9 +51,12 @@
 	private static final String ACTION_GET_COURSES = "getCoursesPublished";
 	private static final String ACTION_GET_TERMS = "getEnrollmentTerms";
 	private static final String EMAIL_SUFFIX="@umich.edu";
+	private static final String ACTION_POLLING = "polling";
+	private static final String WORKING = "working";
 	private String canvasToken=CanvasCourseReportFilter.canvasToken;
 	private String canvasURL=CanvasCourseReportFilter.canvasURL;
 	ResourceBundle props = ResourceBundle.getBundle("coursereport");
+	ConcurrentHashMap<Thread,CourseReportTask> courseReportReqThreads=new ConcurrentHashMap<Thread,CourseReportTask>();
```

Comment:
Will Document that if run on more than on server 1) Make the load balancer session sticky 2) Revisit this issue